provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,271 @@
+            this._pollingTimer = null;
+            this._currentHttpRequest = null;
+            this._restApiClient = null;
+            this._body = null;
+            this._forceRegistration = false;
+            this._operationId = '';
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            this._forceRegistration = forceRegistration;
+            this._body = body;
+            this._restApiClient = new RestApiClient({ 'host' : this._provisioningServer , 'sharedAccessSignature' : authorization}, 'azure-node-provisioning', httpBase);
+            this._fsm.transition('registering', callback);
+          },
+          disconnect : (callback) => {
+            // nothing to do.
+            this._safeCallback(callback);
+          }
+        },
+        registering: {
+          _onEnter: (callback) => {
+            this._sendRegistrationRequest((err, responseBody) => {
+              if (err) {
+                this._safeCallback(callback, err);
+                this._fsm.transition('idle');
+              } else if (this._isAssigning(responseBody)) {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_016: [ If the registration response has a success code with a 'status' of 'Assigning', `register` shall fire an `operationStatus` event with the deserialized body ] */
+                  this.emit('operationStatus', responseBody);
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_017: [ If the registration response has a success code with a 'status' of 'Assigning', `register` shall start polling for operation updates ] */
+                  this._operationId = this._getOperationId(responseBody);
+                  this._fsm.transition('waiting_to_poll', callback);
+              } else if (this._isAssigned(responseBody)) {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_015: [ If the registration response has a success code with a 'status' of 'Assigned', `register` call the `callback` with `err` == `null` and result `containing` the deserialized body ] */
+                callback(null, responseBody);
+                this._fsm.transition('idle');
+              } else {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_029: [ If the registration response has a success code with a 'status' that is not 'Assigned' or  'Assigning', `register` shall call the callback with a `SyntaxError` error. ] */
+                this._safeCallback(callback, new SyntaxError('badly formed registration response'));
+                this._fsm.transition('idle');
+              }
+            });
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
```

Comment:
>callback(new errors.InvalidOperationError('registration already in progress.')); [](start = 12, length = 80)

missed that requirement earlier - should this operation be idempotent if the values are the same or on the contrary cancel and start a new one? just thinking of potential retry logic in the future?

Reply:
Good question, but I don't see any reason right now to change this.  

---
In reply to: [141481719](https://github.com/Azure/azure-iot-sdk-node/pull/93#discussion_r141481719) [](ancestors = 141481719)